### PR TITLE
API provides functionality for visitor to view all listed articles

### PR DESCRIPTION
**Pull request by JaimeCrz**
# [PT Story](https://www.pivotaltracker.com/story/show/171811166)

```
As a Visitor 
In order to choose an article to read 
I would like to be able to see the articles listed on a page
```

## Changes proposed in this pull request:
* Adds unit test for Articles model, and rspec validations
* Adds unit test for Articles controller, and request spec
* Installs and setups rake-cors

## What I have learned working on this feature:
* Refreshed  knowledge about, creating Models and Controllers in Rails.
* Refreshed knowledge about  rails routes.
* Refreshed knowledge about implementing the rake-cors gem.
* Refreshed knowledge about delete unnecessary github cache files.

## Gems added
-rack-cors

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,7 @@
 !/tmp/.keep
 coverage/
 
+
```
Comment (blake-futchi) [inline .gitignore]: delete line

File: app/controllers/api/articles_controller.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Api::ArticlesController < ApplicationController
+    def index
+    collection_articles = Article.all
```
Comment (blake-futchi) [inline app/controllers/api/articles_controller.rb:3]: indentation + 1 on line 3

File: app/models/article.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class Article < ApplicationRecord
+    validates_presence_of :title, :teaser, :content
```
Comment (blake-futchi) [inline app/models/article.rb]: format indentation

File: db/migrate/20200319134722_create_articles.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class CreateArticles < ActiveRecord::Migration[6.0]
+  def change
+    create_table :articles do |t|
+      t.string :title
+      t.text :teaser
+      t.text :content
+
```
Comment (blake-futchi) [inline db/migrate/20200319134722_create_articles.rb:7]: do we get id added to db column once we use serializer?

File: spec/models/article_spec.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    it { is_expected.to have_db_column :content }
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :title}
+    it { is_expected.to validate_presence_of :teaser }
+    it { is_expected.to validate_presence_of :content } 
+  end
+
+  describe "Check if new article is created" do
```
Comment (blake-futchi) [inline spec/models/article_spec.rb]: johan and I are wondering if this test should be in the other spec file?

Comment (JaimeCrz) [reply spec/models/article_spec.rb]: done

File: db/migrate/20200319134722_create_articles.rb (hunk 4 above)
Comment (JaimeCrz) [reply db/migrate/20200319134722_create_articles.rb:7]: Rails gives auto Id to the items check the response file we sent to you earlier while we had the meeting: (tagged you)

File: app/models/article.rb (hunk 3 above)
Comment (JaimeCrz) [reply app/models/article.rb]: done, tabbing op :dancer:

File: app/controllers/api/articles_controller.rb (hunk 2 above)
Comment (JaimeCrz) [reply app/controllers/api/articles_controller.rb:3]: fixed

File: .gitignore (hunk 1 above)
Comment (JaimeCrz) [reply .gitignore]: done!

**Pull request merged**